--- a/zfs_stats_	2017-01-26 20:40:22.000000000 -0500
+++ b/zfs_stats_	2017-10-29 15:56:25.713120469 -0400
@@ -46,7 +46,8 @@
 PREFETCH_METADATA_MISSES=`echo "$CONTENT" | grep "^prefetch_metadata_misses" | awk '{print $3;}'`
 
 SIZE=`echo "$CONTENT" | grep "^size" | awk '{print $3;}'`
-MRU_SIZE=`echo "$CONTENT" | grep "^p\s" | awk '{print $3;}'`
+MRU_SIZE=`echo "$CONTENT" | grep "^mru_size" | awk '{print $3;}'`
+MFU_SIZE=`echo "$CONTENT" | grep "^mfu_size" | awk '{print $3;}'`
 MAX_SIZE=`echo "$CONTENT" | grep "^c_max" | awk '{print $3;}'`
 MIN_SIZE=`echo "$CONTENT" | grep "^c_min" | awk '{print $3;}'`
 TARGET_SIZE=`echo "$CONTENT" | grep "^c\s" | awk '{print $3;}'`
@@ -88,12 +89,6 @@
 PREFETCH_METADATA_HIT_PERC=`echo "scale=2 ; (100*$PREFETCH_METADATA_HITS/$ARC_HITS)" | $BC`
 PREFETCH_METADATA_MISSES_PERC=`echo "scale=2 ; (100*$PREFETCH_METADATA_MISSES/$ARC_MISSES)" | $BC`
 
-if [ $SIZE -gt $TARGET_SIZE ]; then
-	MFU_SIZE=`echo "$SIZE-$MRU_SIZE" | $BC`
-else
-	MFU_SIZE=`echo "$TARGET_SIZE-$MRU_SIZE" | $BC`
-fi
-
 L2_ACCESSES_TOTAL=`echo "$L2_HITS+$L2_MISSES" | $BC`
 if [ $L2_ACCESSES_TOTAL -gt 0 ]; then
 	L2_HIT_RATIO_PERC=`echo "scale=2 ; (100*$L2_HITS/$L2_ACCESSES_TOTAL)" | $BC`
